feature/interface/aggregate/otg_tests/aggregate_forwarding_viable_test/aggregate_forwarding_viable_test.go

```diff
@@ -0,0 +1,596 @@
+	rxPackets := float32(recvMetric.GetCounters().GetInPkts())
+	lostPackets := txPackets - rxPackets
+	lossPct := lostPackets * 100 / txPackets
+
+	if got := lossPct; got > 0 {
+		t.Fatalf("LossPct for flow %s: got %g, want 0", flow.Name(), got)
+	}
+}
+
+// verifyCounterDiff finds the difference between counter values before and after sending traffic. It also calculates if there is any packet loss.
+func (tc *testArgs) verifyCounterDiff(t *testing.T, before, after []*oc.Interface_Counters, want []float64) {
+	b := &strings.Builder{}
+	w := tabwriter.NewWriter(b, 0, 0, 1, ' ', 0)
+	approxOpt := cmpopts.EquateApprox(0 /* frac */, 0.01 /* absolute */)
+	fmt.Fprint(w, "Interface Counter Deltas\n\n")
+	fmt.Fprint(w, "Name\tInPkts\tInOctets\tOutPkts\tOutOctets\n")
+	allOutPkts := []uint64{}
+
+	for i, port := range tc.dutPorts[1:] {
+		inPkts := after[i].GetInUnicastPkts() - before[i].GetInUnicastPkts()
+		inOctets := after[i].GetInOctets() - before[i].GetInOctets()
+		outPkts := after[i].GetOutUnicastPkts() - before[i].GetOutUnicastPkts()
+		allOutPkts = append(allOutPkts, outPkts)
+		outOctets := after[i].GetOutOctets() - before[i].GetOutOctets()
+
+		fmt.Fprintf(w, "%s\t%d\t%d\t%d\t%d\n",
+			port,
+			inPkts, inOctets,
+			outPkts, outOctets)
+	}
+	got, _ := normalize(allOutPkts)
+
+	t.Logf("outPkts normalized got: %v", got)
+	t.Logf("want: %v", want)
+	t.Run("Ratio", func(t *testing.T) {
+		if diff := cmp.Diff(want, got, approxOpt); diff != "" {
+			t.Errorf("Packet distribution ratios -want,+got:\n%s", diff)
+		}
+	})
+
+	w.Flush()
+	t.Log(b)
+}
+
+// testAggregateForwardingFlow set the forwardingViable tag based on test case and returns test results of packet distribution and packet loss for each LAG type.
+// Interfaces that are not viable for forwarding should still be allowed to receive traffic, but should not be used for sending packets.
+func (tc *testArgs) testAggregateForwardingFlow(t *testing.T, forwardingViable bool) {
+	var want []float64
+	lp := newLinkPairs(tc.dut, tc.ate)
+	// Update the interface config of one port in port-channel when the forwarding flag is set as false.
+	if !forwardingViable {
+		t.Log("First port does not forward traffic because it is marked as not viable.")
+		gnmi.Update(t, tc.dut, gnmi.OC().Interface(tc.dutPorts[1].Name()).ForwardingViable().Config(), forwardingViable)
+	}
+
+	i1 := ateSrc.Name
+	i2 := ateDst.Name
+
+	tc.top.Flows().Clear().Items()
+	flow := tc.top.Flows().Add().SetName("flow")
+	flow.Metrics().SetEnable(true)
+	flow.Size().SetFixed(128)
+	flow.Packet().Add().Ethernet().Src().SetValue(ateSrc.MAC)
+
+	flow.TxRx().Device().SetTxNames([]string{i1 + ".IPv4"}).SetRxNames([]string{i2 + ".IPv4"})
+	v4 := flow.Packet().Add().Ipv4()
+	v4.Src().SetValue(ateSrc.IPv4)
+	v4.Dst().SetValue(ateDst.IPv4)
+	tcp := flow.Packet().Add().Tcp()
+	tcp.SrcPort().SetValues(generateRandomPortList(65534))
+	tc.ate.OTG().PushConfig(t, tc.top)
+	tc.ate.OTG().StartProtocols(t)
```

Comment:
Do we want to use otgutils.WaitForARP here?

Reply:
That we use only when creating "port" instead of "device" based flows, because with the latter the ARP is automatically resolved.